src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,1184 @@
+#       --string-arg ota_image:${SU_OTA_REQUESTOR_V2}
+#       --int-arg ota_provider_port:5541
+#     factory-reset: true
+#     quiet: false
+# === END CI TEST ARGUMENTS ===
+
+import asyncio
+import logging
+import queue
+import time
+
+from mobly import asserts
+from TC_SUTestBase import SoftwareUpdateBaseTest
+
+import matter.clusters as Clusters
+from matter import ChipDeviceCtrl
+from matter.interaction_model import Status
+from matter.testing.event_attribute_reporting import AttributeMatcher, AttributeSubscriptionHandler, EventSubscriptionHandler
+from matter.testing.matter_testing import TestStep, async_test_body, default_matter_test_main
+
+# Create a logger
+logger = logging.getLogger(__name__)
+
+
+class TC_SU_2_2(SoftwareUpdateBaseTest):
+
+    async def add_single_ota_provider_if_missing(self, controller, requestor_node_id: int, provider_node_id: int):
+        """
+        Adds a single OTA provider to the Requestor DefaultOTAProviders attribute
+        only if no provider is currently registered. If a provider already exists,
+        the function oreserve the current DefaultOTAProviders.
+
+        Args:
+            controller: The device controller.
+            requestor_node_id (int): Node ID of the Requestor device.
+            provider_node_id (int): Node ID of the OTA Provider to add.
+
+        Returns:
+            None
+        """
+
+        # Read existing DefaultOTAProviders on the Requestor
+        current_providers = await self.read_single_attribute_check_success(
+            dev_ctrl=controller,
+            cluster=Clusters.OtaSoftwareUpdateRequestor,
+            attribute=Clusters.OtaSoftwareUpdateRequestor.Attributes.DefaultOTAProviders
+        )
+        logger.info(f'Prerequisite #4.0 - Current DefaultOTAProviders on Requestor: {current_providers}')
+
+        # If there is already a provider, skip adding
+        if current_providers:
+            logger.info(f"Skipping add: Requestor already has providers {current_providers}")
+            return
+
+        await self.set_default_ota_providers_list(controller, provider_node_id, requestor_node_id)
+        logger.info("Prerequisite #4.0 - Write DefaultOTAProviders completed.")
+
+    async def setup_provider(self, provider_app_path: str, ota_image_path: str, extra_args: list[str]):
+        """
+        Launch an OTA Provider process with the specified image and extra arguments.
+
+        Args:
+            provider_app_path (str): Path to the OTA provider application executable.
+            ota_image_path (str): Path to the OTA image file.
+            extra_args (list[str]): Additional command-line arguments to pass to the provider.
+
+        Returns:
+            None
+        """
+
+        logger.info(f"""Prerequisite - Provider info:
+            provider_app_path: {provider_app_path},
+            ota_image_path: {ota_image_path},
+            discriminator: {self.provider_data["discriminator"]},
```

Comment:
Please don't use string keys for passing parameters. If required, use a dataclass.